src/bosh.js

```diff
@@ -767,7 +767,9 @@ Strophe.Bosh.prototype = {
                 // expanding retry window
                 var backoff = Math.min(Math.floor(Strophe.TIMEOUT * this.wait),
                                        Math.pow(req.sends, 3)) * 1000;
-                setTimeout(sendFunc, backoff);
+                setTimeout(function() {
+                    sendFunc();
+                }, backoff);
```

Comment:
Hi @jcbrand 

Yes, this and everything in commit 23974bc is about the reported setTimeout issues.

I.e:
```bash
Warning: `setTimeout` called in potentially dangerous manner
In order to prevent vulnerabilities, the `setTimeout` and `setInterval` functions should be called only with function expressions as their first argument.
	Automated signing severity: high
	Suggestions for passing automated signing:
Please do not ever call `setTimeout` or `setInterval` with string arguments. If you are passing a function which is not being correctly detected as such, please consider passing a closure or arrow function, which in turn calls the original function.```